gcc/rust/expand/rust-macro-builtins.cc

```diff
@@ -74,8 +74,7 @@ const BiMap<std::string, BuiltinMacro> MacroBuiltin::builtins = {{
 
 }};
 
-std::unordered_map<
-  std::string, std::function<AST::Fragment (Location, AST::MacroInvocData &)>>
+std::unordered_map<std::string, AST::transcriber_t>
```

Comment:
...here. It feels a bit weird to have the `transcriber_t` type in the AST namespace. `AST::MacroRulesDefinition::transcriber_t` would be clearer IMO. But that's very nitpicky

Reply:
You're correct... And that was initially my plan, but there's a circular dep here, and it's not possible to use incomplete types in templates... macro includes fragment, so declaring the type alias after the `Fragment` class def is the easy way. Or maybe there's a C++ trick for that, but I don't think so. I'll see if there's another class defined after both Fragment and MacroInvocData that could be a good candidate to host this type.

Reply:
I've looked around and didn't see any obvious candidate. Will give it another try later this week...

Reply:
ah, didn't think about the circular dependency :) that's completely fine with me then!